version.js

```diff
@@ -25,19 +68,34 @@ async function createRelease(version, releaseNotes, config) {
 async function getCurrentVersion(config) {
     const data = await config.octokit.git.listMatchingRefs({
         ...github.context.repo,
-        namespace: 'tags/',
+        ref: 'tags/',
+    })
+
+    const objectsByVersion = new Map()
+    const versions = []
+
+    data.data.forEach((ref) => {
+        const version = semver.parse(ref.ref.replace(/^refs\/tags\//g, ''), { loose: true })
+
+        if (version !== null) {
+            objectsByVersion[version] = ref.object
+            versions.push(version)
+        }
     })
 
-    const versions = data.data
-        .map((ref) => semver.parse(ref.ref.replace(/^refs\/tags\//g, ''), { loose: true }))
-        .filter((version) => version !== null)
-        .sort(semver.rcompare)
+    versions.sort(semver.rcompare)
+
+    if (config.baseBranch) {
+        const branch = process.env.GITHUB_BASE_REF || (process.env.GITHUB_REF && process.env.GITHUB_REF.split('/')[2])
```

Comment:
In the context of a Pull Request
* `GITHUB_BASE_REF` is set to branch name that the PR targets
* `GITHUB_REF` is set to `refs/pull/<pr number>/merge`

On merge (merge, squash, and rebase)
* `GITHUB_BASE_REF` is not set
* `GITHUB_REF` is `refs/heads/<branch name>`